http_parser.c

```diff
@@ -1782,9 +1782,9 @@ size_t http_parser_execute (http_parser *parser,
 
         if (parser->flags & F_TRAILING) {
           /* End of a chunked request */
-          UPDATE_STATE(NEW_MESSAGE());
-          CALLBACK_NOTIFY(message_complete);
-          break;
+          UPDATE_STATE(s_message_done);
```

Comment:
I'm not the original author (just doing the selective merge), but I figure this change here is necessary to get the chunk_complete event out at the right time. Are you suggesting an alternative simpler solution? I can't think of one.